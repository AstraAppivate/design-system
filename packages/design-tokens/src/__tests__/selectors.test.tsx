import React from 'react'
import styled from 'styled-components'
import { render, RenderResult } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { selectors } from '..'

const { mediaQuery, mq } = selectors

describe('mediaQuery / mq', () => {
  let wrapper: RenderResult

  describe('with invalid breakpoints', () => {
    it('throws an error', () => {
      expect(() => {
        // @ts-ignore
        const css = mq({ gte: 'foo', lt: 'bar' })`
          background-color: white;
        `
      }).toThrowError(/Invalid breakpoints/)
    })
  })

  describe('with valid breakpoints', () => {
    const StyledComponent = styled.div`
      background-color: white;
      font-size: initial;

      ${mediaQuery({ gte: 'xs' })`
        background-color: blue;
      `}

      ${mq({ gte: 's', lt: 'm' })`
        background-color: green;
      `}

      ${mq({ gte: 'l', lt: 'xl', media: '' })`
        background-color: black;
      `}
    `

    beforeEach(() => {
      wrapper = render(<StyledComponent data-testid="test-element" />)
    })

    it('snapshot test', () => {
      // NOTE: We usually advise against snapshot testing
      const { container } = wrapper
      expect(container.firstChild).toMatchSnapshot()
    })

    it('should set styles correctly for each breakpoint', () => {
      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'background-color',
        'white'
      )

      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'background-color',
        'blue',
        {
          media: 'only screen and (min-width:576px)',
        }
      )
      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'background-color',
        'green',
        {
          media: 'only screen and (min-width:768px) and (max-width:1024px)',
        }
      )

      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'background-color',
        'black',
        {
          media: '(min-width:1200px) and (max-width:1400px)',
        }
      )
    })

    it('should set the `font-size` css property correctly', () => {
      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'font-size',
        'initial'
      )

      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'font-size',
        '94%',
        {
          media: 'only screen and (min-width:576px)',
        }
      )

      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'font-size',
        '96%',
        {
          media: 'only screen and (min-width:768px) and (max-width:1024px)',
        }
      )

      expect(wrapper.getByTestId('test-element')).toHaveStyleRule(
        'font-size',
        '100%',
        {
          media: '(min-width:1200px) and (max-width:1400px)',
        }
      )
    })
  })
})
