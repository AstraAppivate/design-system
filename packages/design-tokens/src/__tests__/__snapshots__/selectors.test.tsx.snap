// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mediaQuery / mq with valid breakpoints snapshot test 1`] = `
.c0 {
  background-color: white;
  font-size: initial;
}

@media only screen and (min-width:576px) {
  .c0 {
    font-size: 94%;
    background-color: blue;
  }
}

@media only screen and (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 96%;
    background-color: green;
  }
}

@media (min-width:1200px) and (max-width:1400px) {
  .c0 {
    font-size: 100%;
    background-color: black;
  }
}

<div
  class="c0"
  data-testid="test-element"
/>
`;
